File: frontend/src/types/analytics.ts
// ─── Analytics Types ─────────────────────────────────────────────────────────

// ─── Overview ────────────────────────────────────────────────────────────────

export interface WeekOverWeek {
  connectionsSent: number
  messagesSent: number
}

export interface OverviewMetrics {
  connectionsSent: number
  connectionsAccepted: number
  acceptanceRate: number
  messagesSent: number
  repliesReceived: number
  replyRate: number
  totalLeads: number
  activeCampaigns: number
  outreachScore: number
  weekOverWeek: WeekOverWeek
}

// ─── Funnel ──────────────────────────────────────────────────────────────────

export interface FunnelStep {
  label: string
  value: number
  pct: number
  color: string
}

// ─── Daily Activity ──────────────────────────────────────────────────────────

export interface DailyActivity {
  date: string
  label: string
  connectionsSent: number
  messagesSent: number
  replies: number
  total: number
}

// ─── Campaign Row ────────────────────────────────────────────────────────────

export interface CampaignRow {
  id: string
  name: string
  status: string
  totalLeads: number
  connectionsSent: number
  acceptanceRate: number
  messagesSent: number
  replyRate: number
}

// ─── Activity Event ──────────────────────────────────────────────────────────

export interface ActivityEvent {
  id: string
  type: string
  status: string
  leadName: string | null
  campaignName: string
  time: string
}

// ─── Top-Level Response ──────────────────────────────────────────────────────

export interface AnalyticsData {
  overview: OverviewMetrics
  funnel: FunnelStep[]
  dailyActivity: DailyActivity[]
  campaigns: CampaignRow[]
  recentActivity: ActivityEvent[]
}

// ─── Config Constants ────────────────────────────────────────────────────────

export const ACTIVITY_META: Record<string, { emoji: string; label: string }> = {
  connection_request: { emoji: '🔗', label: 'Connection request sent to' },
  message_sent:       { emoji: '💬', label: 'Message sent to' },
  reply_received:     { emoji: '✅', label: 'Reply received from' },
  profile_view:       { emoji: '👁️', label: 'Viewed profile of' },
  follow:             { emoji: '➕', label: 'Followed' },
  like_post:          { emoji: '👍', label: 'Liked a post by' },
  inmail_sent:        { emoji: '📩', label: 'InMail sent to' },
  email_sent:         { emoji: '📧', label: 'Email sent to' },
}

export const STATUS_BADGE: Record<string, { label: string; cls: string }> = {
  active:    { label: 'Active',   cls: 'bg-green-100 text-green-700' },
  running:   { label: 'Running',  cls: 'bg-green-100 text-green-700' },
  paused:    { label: 'Paused',   cls: 'bg-yellow-100 text-yellow-700' },
  draft:     { label: 'Draft',    cls: 'bg-gray-100 text-gray-600' },
  completed: { label: 'Done',     cls: 'bg-blue-100 text-blue-700' },
  canceled:  { label: 'Canceled', cls: 'bg-red-100 text-red-700' },
}

File: backend/internal/models/analytics.go
package models

// ── Analytics Response DTOs ─────────────────────────────────────────────────

// WeekOverWeek holds percentage changes compared to the previous week.
type WeekOverWeek struct {
	ConnectionsSent float64 `json:"connectionsSent"`
	MessagesSent    float64 `json:"messagesSent"`
}

// OverviewMetrics aggregates top-level outreach performance numbers.
type OverviewMetrics struct {
	ConnectionsSent     int          `json:"connectionsSent"`
	ConnectionsAccepted int          `json:"connectionsAccepted"`
	AcceptanceRate      float64      `json:"acceptanceRate"`
	MessagesSent        int          `json:"messagesSent"`
	RepliesReceived     int          `json:"repliesReceived"`
	ReplyRate           float64      `json:"replyRate"`
	TotalLeads          int          `json:"totalLeads"`
	ActiveCampaigns     int          `json:"activeCampaigns"`
	OutreachScore       int          `json:"outreachScore"`
	WeekOverWeek        WeekOverWeek `json:"weekOverWeek"`
}

// FunnelStep represents one tier of the outreach funnel.
type FunnelStep struct {
	Label string `json:"label"`
	Value int    `json:"value"`
	Pct   int    `json:"pct"`
	Color string `json:"color"`
}

// DailyActivity holds per-day activity counts for the 14-day chart.
type DailyActivity struct {
	Date            string `json:"date"`
	Label           string `json:"label"`
	ConnectionsSent int    `json:"connectionsSent"`
	MessagesSent    int    `json:"messagesSent"`
	Replies         int    `json:"replies"`
	Total           int    `json:"total"`
}

// CampaignRow is a simplified campaign for the analytics performance table.
type CampaignRow struct {
	ID              string  `json:"id"`
	Name            string  `json:"name"`
	Status          string  `json:"status"`
	TotalLeads      int     `json:"totalLeads"`
	ConnectionsSent int     `json:"connectionsSent"`
	AcceptanceRate  float64 `json:"acceptanceRate"`
	MessagesSent    int     `json:"messagesSent"`
	ReplyRate       float64 `json:"replyRate"`
}

// ActivityEvent represents a single entry in the recent activity feed.
type ActivityEvent struct {
	ID           string  `json:"id"`
	Type         string  `json:"type"`
	Status       string  `json:"status"`
	LeadName     *string `json:"leadName"`
	CampaignName string  `json:"campaignName"`
	Time         string  `json:"time"`
}

// AnalyticsData is the top-level response returned by GET /api/analytics.
type AnalyticsData struct {
	Overview       OverviewMetrics `json:"overview"`
	Funnel         []FunnelStep    `json:"funnel"`
	DailyActivity  []DailyActivity `json:"dailyActivity"`
	Campaigns      []CampaignRow   `json:"campaigns"`
	RecentActivity []ActivityEvent `json:"recentActivity"`
}
